addons/point_of_sale/models/pos_order.py

```diff
@@ -1002,14 +1002,199 @@ def action_receipt_to_customer(self, name, client, ticket):
+            if order['pricelist_id']:
+                order['pricelist_id'] = order['pricelist_id'][0]
+            if order['partner_id']:
+                order['partner_id'] = order['partner_id'][0]
+
+            if not 'lines' in order:
+                order['lines'] = []
+            if not 'statement_ids' in order:
+                order['statement_ids'] = []
+
+            del order['id']
+            del order['session_id']
+            del order['pos_reference']
+            del order['create_date']
+
+        return orders
+
+    @api.model
+    def remove_from_ui(self, server_ids):
+        """ Remove orders from the frontend PoS application
+
+        Remove orders from the server by id.
+        :param server_ids: list of the id's of orders to remove from the server.
+        :type server_ids: list.
+        :returns: list -- list of db-ids for the removed orders.
+        """
+        orders = self.search([('id', 'in', server_ids), ('state', '=', 'draft')])
+        orders.write({'state': 'cancel'})
+        # TODO Looks like delete cascade is a better solution.
+        orders.mapped('payment_ids').sudo().unlink()
+        orders.sudo().unlink()
+        return orders.ids
+
     @api.model
     def search_paid_order_ids(self, config_id, domain, limit, offset):
         """Search for 'paid' orders that satisfy the given domain, limit and offset."""
-        default_domain = ['&', ('config_id', '=', config_id), '!', '|', ('state', '=', 'draft'), ('state', '=', 'cancelled')]
-        real_domain = AND([domain, default_domain])
-        ids = self.search(AND([domain, default_domain]), limit=limit, offset=offset).ids
+        default_domain = ['!', '|', ('state', '=', 'draft'), ('state', '=', 'cancelled')]
+        if domain == []:
+            real_domain = AND([[['config_id', '=', config_id]], default_domain])
+        else:
+            real_domain = AND([domain, default_domain])
+        orders = self.search(real_domain, limit=limit, offset=offset)
+        orderlines = self.env['pos.order.line'].search(['|', ('refunded_orderline_id.order_id', 'in', orders.ids), ('order_id', 'in', orders.ids)])
+        ordersInfo = {}
+        for orderline in orderlines:
+            if orderline.order_id.id in orders.ids:
+                if orderline.order_id.id not in ordersInfo:
+                    ordersInfo.setdefault(orderline.order_id.id, orderline.write_date)
+                elif ordersInfo[orderline.order_id.id] < orderline.write_date:
+                    ordersInfo[orderline.order_id.id] = orderline.write_date
+            else:
+                if orderline.refunded_orderline_id.order_id.id not in ordersInfo:
+                    ordersInfo.setdefault(orderline.refunded_orderline_id.order_id.id, orderline.write_date)
+                elif ordersInfo[orderline.refunded_orderline_id.order_id.id] < orderline.write_date:
+                    ordersInfo[orderline.refunded_orderline_id.order_id.id] = orderline.write_date
```

Comment:
This code seems alike the one above, you might just create a variable to store the key which can either be normal or refunded